src/common/camera_control.c

```diff
@@ -1057,12 +1057,12 @@ static gboolean dt_camctl_update_cameras(const dt_camctl_t *c)
 void *dt_update_cameras_thread(void *ptr)
 {
   dt_pthread_setname("gphoto_update");
-  /* make sure control is up and running */
-  for(int k = 0; k < 100; k++)
-  {
-    if(dt_control_running()) break;
-    g_usleep(100000);
-  }
+
+  dt_control_t *control = darktable.control;
+  dt_atomic_add_int(&control->running_jobs, 1);
+  g_usleep(100000); // no haste, keep going in main thread
```

Comment:
I'd recommend adding another zero to make it a full second, to be extra-sure that we don't accidentally skip the run loop because it took a while for `dt_control_running` to actually return TRUE.  As the comment says, no haste....